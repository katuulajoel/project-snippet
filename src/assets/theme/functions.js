export const pxToRem = (pixels) => {
  const value = 1 * (pixels / 16);
  return `${value}rem`;
};

export const functions = {
  pxToRem,
};
