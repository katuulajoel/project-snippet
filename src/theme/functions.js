export const pxToRem = (pixels) => {
  const value = 1 * (pixels / 16);
  return `${value}rem`;
};

const pxToLineHeight = (pixels) => {
  const value = pixels / 16;
  return `${value}rem`;
};

export const functions = {
  pxToRem,
  pxToLineHeight,
};
