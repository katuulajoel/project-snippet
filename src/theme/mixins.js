import { css } from "styled-components";

const clearfix = css`
  &::after {
    display: block;
    clear: both;
    content: "";
  }
`;

const link = (color) => `
    color: ${color};

    &:hover, &:focus, &:active {
        color: ${color};
        text-decoration: none;
    }
`;

const scrollbar = css`
  /* width */
  &::-webkit-scrollbar {
    width: 14px;
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(143, 155, 179, 0.15);
    box-shadow: 0px 4px 8px rgba(62, 72, 87, 0.04);
    background-clip: padding-box;
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
    background-clip: padding-box;
    background-color: #8f9bb3;
    box-shadow: -1px -1px 0px rgba(62, 72, 87, 0.05);
    -webkit-box-shadow: inset -1px -1px 0px rgba(62, 72, 87, 0.05),
      inset 1px 1px 0px rgba(62, 72, 87, 0.05);
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    border: 10px solid rgba(0, 0, 0, 0);
    border-top: none;
    border-left: none;
    background-clip: padding-box;
    background-color: #6b7588;
    cursor: pointer;
  }
`;

const borderBottom = css`
  border-bottom: 1px solid #062e64;
`;

export const mixins = {
  clearfix,
  link,
  borderBottom,
  scrollbar,
};
