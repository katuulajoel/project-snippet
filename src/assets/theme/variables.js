import { pxToRem } from "./functions";

const colors = {
  "dark-blue": "#062E64",
  gray: "#8F9BB3",
  "text-black": "#151A30",
  "text-black-normal": "#3E4857",
  "bg-color": "#f8f8f8",
};

const fontSizes = {
  "font-size-normal": pxToRem(14),
  "font-size-sm": pxToRem(14 * 0.875),
  "font-size-lg": pxToRem(14 * 1.25),
  "font-size-xl": pxToRem(14 * 1.5),
  "font-size-xxl": pxToRem(14 * 1.75),
};

export const variables = {
  colors,
  fontSizes,
};
