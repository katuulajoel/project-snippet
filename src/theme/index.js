import { functions } from "./functions";
import { mixins } from "./mixins";
import { variables } from "./variables";

export default {
  ...variables,
  mixins: mixins,
  functions: functions,
};
