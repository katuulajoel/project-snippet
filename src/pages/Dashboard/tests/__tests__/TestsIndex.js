import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ThemeProvider } from "styled-components";
import Tests from "../index";
import theme from "../../../../theme";

const middlewares = [thunk];

const mockAppState = {
  Invoice: {
    isMakingRequest: {},
    errors: {},
    summary: {},
    list: { data: [], count: 0, next: "", previous: "" },
    invoice: {},
    csv: {},
  },
  TestResults: {
    count: {},
    errors: { fetch: null },
    isFetching: { "6LMlpGnA": true, default: false, selectionKey: "6LMlpGnA" },
    isSaved: {},
    isSaving: {},
    next: {},
    previous: {},
    results: [],
    selectedFilters: [],
  },
};

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

describe("Dashboard test", () => {
  it("Should match snapshot test", () => {
    global.URL.createObjectURL = jest.fn(() => "details");
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={mockAppStore(mockAppState)}>
            <ThemeProvider theme={theme}>
              <Tests />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
