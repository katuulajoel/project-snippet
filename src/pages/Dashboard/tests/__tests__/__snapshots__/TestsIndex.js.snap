// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard test Should match snapshot test 1`] = `
Array [
  <nav
    className="sc-iCoGMd wWrIP navbar "
  >
    <div
      className="title-bar"
    >
      <a
        className="navbar-brand"
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
      <ul
        className="navbar-nav ml-auto"
      >
        <li>
          <div
            className="input-group sc-hKFxyN jgJWoP input-search"
          >
            <span
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                <i
                  className="tg-ic-search1  "
                />
              </span>
            </span>
            <div
              className="sc-bdnxRM lboORq"
              data-testid="input"
            >
              <input
                autoComplete="off"
                className="form-control"
                data-testid=""
                id=""
                label="Label"
                name="search"
                onChange={[Function]}
                placeholder="Search...."
                type="input"
                value=""
              />
            </div>
            <span
              className="input-group-prepend"
            />
          </div>
          
        </li>
      </ul>
    </div>
    <div
      className="sc-gKAaRy jFGIs"
    />
  </nav>,
  <div
    className="sc-pNWdM ibuYVp sidebar"
    id="sidebar"
  >
    <img
      alt="company-logo"
      src={Object {}}
    />
    <ul
      className="nav"
    >
      <li>
        <a
          aria-current={null}
          href="/dashboard"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-outline-dashboard tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Dashboard
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/projects"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-baseline-folder-open tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Projects
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/payments/in/all"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-round-payment tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Payments
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/settings"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-outline-settings tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Settings
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/community"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-baseline-folder-open tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Community
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          data-testid="signout"
          href="/logout"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-log-out tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Log Out
          </span>
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="sc-jrsJWt fuWeUk"
  >
    <div>
      <div>
        <div
          className="sc-kEqXSa cMBdJe"
        >
          <div
            className="filter-links"
          />
          <h3>
            Results
          </h3>
          <div
            className="sc-dIvrsQ eDFvmN"
            style={
              Object {
                "float": "right",
              }
            }
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <i
                className="tg-ic-filter-variant tunga-ic-sz-sm "
              />
               Filter
               
            </a>
            <div
              className="input-group sc-hKFxyN jgJWoP input-search"
            >
              <span
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  <i
                    className="tg-ic-search1  "
                  />
                </span>
              </span>
              <div
                className="sc-bdnxRM lboORq"
                data-testid="input"
              >
                <input
                  autoComplete="off"
                  className="form-control"
                  data-testid=""
                  id=""
                  label="Label"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search...."
                  type="input"
                  value=""
                />
              </div>
              <span
                className="input-group-prepend"
              />
            </div>
            
          </div>
        </div>
      </div>
      <div
        className="loading"
        data-testid="loading"
      >
        <img
          src={Object {}}
          style={
            Object {
              "width": "20px",
            }
          }
        />
         
        
      </div>
    </div>
  </div>,
]
`;
